### ISSUE TITLE
Nullability annotations aren't reliably working

### ISSUE THREAD

**Issue by stuartmorgan** [dart-lang/native#1813]
I just updated my `video_player` prototype WIP to the new `jnigen` 0.13.0, and I'm seeing some incorrect output where I have nullability annotations. For example:

```java
public final class VideoPlayer implements TextureRegistry.SurfaceProducer.Callback {
  ...

  public static @NonNull VideoPlayer create(...
```

The result of `jnigen` is:
```dart
static VideoPlayer? create
```

Which incorrectly has a nullable return value. (It was originally `@NonNull public static VideoPlayer`, which should also work, but I tried moving it to make sure the location wasn't the problem). I'm seeing the same thing on a bunch of other static factory methods; maybe it's an issue with statics?

**Comment by HosseinYousefi** [comment]
I checked and it seems like the bytecode puts the annotation on the method instead of the method return. Similarly it puts the annotations on the parameters instead of parameter types. I could just add them to the types myself, meaning if a method is annotated with `@NonNull` its return type must be annotated with it. However that doesn't work for arrays as technically `@NonNull String[] foo()` returns an array of non null strings which itself can be null! 

Probably gradle removes some annotation data for release, I'll investigate further...

**Comment by HosseinYousefi** [comment]
OK, it's not about gradle. It's the fact that android's `NonNull` annotation is not general enough, it can only be put on methods, paramaters, fields, ... but not actual types. Meaning that we can't even have `List<@NonNull String>`!

So, maybe what I will do is if the return type is without nullability annotations, use method annotations. The same goes for parameter types.

**Comment by stuartmorgan** [comment]
> Meaning that we can't even have `List<@NonNull String>`!

Yes, we've run into that with Pigeon as well; it's definitely annoying.
